### DISCUSSION TITLE
Understanding OpenThread Code

### DISCUSSION THREAD

**Discussion by cbanier** [openthread/openthread#9959] [Q&A] [answered]
Hello,

I am looking for additional resources to better understand the OpenThread code, beyond what is already available on the official website (https://openthread.io/) and on GitHub. Class diagrams or any other would be especially helpful in grasping the code structure. Are there such documents available, and if so, where can I find them?

Thanks in advanced.
Corentin B.

**Comment by abtink** [comment, marked as answer]
These video may be helpful: [Thread MM April 2022 OpenThread Tutorial](https://www.youtube.com/watch?v=Zlo0V55gJlk&t=12s)

**Reply by cbanier** [reply]
Hello, 
Thank you for these initial responses.

Corentin B.

**Reply by cbanier** [reply]
Good morning,

I need one more clarification. I spent 2/3 weeks trying to understand the core of OT. The reason for that is I would like to develop another kind of header compression (for research purposes). As 6LoWPAN is implemented by default (which makes sense!), I have to modify the core of OT to match my requirements.

Let me know if you have some clue about where I should start ? How to modify the core ? What should I not do ? ...

Many thanks for your help !
Corentin Banier

**Reply by abtink** [reply]
The `Lowpan` class is the module where 6lowpan compression/decompression is implemented. Its methods are mainly used by `MeshForwarder` class.

To streamline future code updates, I would recommend minimizing changes to existing source files.  Instead, prioritize adding new code in separate, dedicated source files. This approach will significantly ease the rebasing process.

**Reply by cbanier** [reply]
Thank you very much for this remark.
